posts/2021-03-25-Rust-1.51.0.md

```diff
@@ -0,0 +1,211 @@
+
+[const-generics-blog]: https://blog.rust-lang.org/2021/02/26/const-generics-mvp-beta.html
+
+### `array::IntoIter` Stabilisation
+As part of const generics stabilising, we're also stabilising a new API that uses it called `array::IntoIter`. `IntoIter` allows you to create a by value iterator over any array. Previously you had to use `[T]`'s (a slice) iterator implementation which returned `&T`, which you then needed to read and copy. Now you can directly iterate over the values without additional copies. *Note:* that `IntoIter::new` may be deprecated once it is possible for arrays to implement `IntoIterator` directly.
+
+```rust
+fn main() {
+  let array = [1, 2, 3, 4, 5];
+  
+  // Previously
+  for item in array.iter().copied() {
+      println!("{}", item);
+  }
+  
+  // Now
+  for item in std::array::IntoIter::new(array) {
+      println!("{}", item);
+  }
+}
+```
+
+### Cargo's New Feature Resolver
+Dependency management is a hard problem, and one of the hardest parts of it is just picking what *version* of a dependency to use when it's depended on by two different packages. This doesn't just include its version number, but also what features are or aren't enabled for the package. Cargo's default behaviour is to merge features for a single package. Let's say you had a dependency called `foo` with features A and B, which was being used by packages `bar` and `baz`, but `bar` depends on `foo+A` and `baz` depends on `foo+B`. Cargo will merge both of those features and compile `foo` as `foo+AB`. This has a benefit that you only have to compile `foo` once, and then it can reused for both `foo` and `bar`.
+
+However, this also comes with a downside. What if `A` and `B` are incompatible? What if your own code is compatible with `foo` but not `foo+A`? A common example of this in the ecosystem is the optional `std` feature included in many `#![no_std]` crates, that allows crates to provide added functionality when `std` is available. Now imagine you want to use the `#![no_std]` version of `foo` in your `#![no_std]` binary, and use the `foo` at build time in your `build.rs`. If your build time dependency depends on `foo+std`, your binary now also depends on `foo+std`, which means it will no longer compile because `std` is not available for your target platform. 
+
+This has been a long-standing issue in cargo, and with this release there's a new `resolver` option in your `Cargo.toml`, where you can set `resolver="2"` to tell cargo to try a new approach to resolving features. You can check out [RFC 2957] for a detailed description of the behaviour, which can be summarised as follows.
```

Comment:
`[RFC 2957]` is missing a link.  https://rust-lang.github.io/rfcs/2957-cargo-features2.html would probably work.